src/NuGet.Services.Staging.Manager/Controllers/PackageController.cs

```diff
@@ -120,16 +123,20 @@ public class PackageController : Controller
                
                 var packageLocations = await v3Service.AddPackage(packageStream, packageMetadata);
 
-                await _stageService.AddPackageToStage(stage, new StagedPackage
-                {
-                    Id = registrationId,
-                    NormalizedVersion = normalizedVersion,
-                    Version = version.ToString(),
-                    UserKey = userKey,
-                    Published = DateTime.UtcNow,
-                    NupkgUrl = packageLocations.Nupkg.ToString(),
-                    NuspecUrl = packageLocations.Nuspec.ToString()
-                });
+                await _stageService.AddPackageToStage(
+                    stage,
+                    new StagedPackage
+                    {
+                        Id = registrationId,
+                        NormalizedVersion = normalizedVersion,
+                        Version = version.ToString(),
+                        UserKey = userKey,
+                        Published = DateTime.UtcNow,
+                        NupkgUrl = packageLocations.Nupkg.ToString(),
+                        NuspecUrl = packageLocations.Nuspec.ToString(),
+                        PackageMetadata = new PackageMetadata().LoadFromNuspec(nuspec, GetUserInformation())

```

Comment:
I wanted to make in an extension method cause I don't want it to be part of NuGet.Service.Staging.Database. Reader is a bit of an overkill..


---
In reply to: [64699046](https://github.com/NuGet/NuGet.Staging/pull/50#discussion_r64699046) [](ancestors = 64699046)